Gems/Atom/RHI/Code/Include/Atom/RHI/Device.h

```diff
@@ -139,6 +139,9 @@ namespace AZ
             //! Notifies after all objects currently in the platform release queue are released
             virtual void ObjectCollectionNotify(RHI::ObjectCollectorNotifyFunction notifyFunction) = 0;
 
+            //! Allows the back-ends to compact SRG related memory if applicable
+            virtual void CompactSRGMemory() {};
```

Comment:
I think it would be better if we return a boolean to indicate if compacting does happen in the function.